src/Cardano/Wallet/Api/Types.hs

```diff
@@ -164,6 +188,27 @@ newtype ApiMnemonicT (sizes :: [Nat]) (purpose :: Symbol) =
                                JSON Instances
 -------------------------------------------------------------------------------}
 
+instance FromJSON Address where
+    parseJSON = genericParseJSON defaultRecordTypeOptions
+instance ToJSON Address where
+    toJSON = genericToJSON defaultRecordTypeOptions
+
+instance FromJSON AddressState where
+    parseJSON = genericParseJSON defaultSumTypeOptions
+instance ToJSON AddressState where
+    toJSON = genericToJSON defaultSumTypeOptions
+
+instance FromJSON (ApiT P.Address) where
+    parseJSON =
+        parseJSON >=>
+          maybe
```

Comment:
I'd drop the pointfree here and use a `do-notation` to makes the reading a bit clearer.

Reply:
Fair enough :)